src/try-path.ts

```diff
@@ -60,15 +59,12 @@ export function getPathsToTry(
   return pathsToTry.length === 0 ? undefined : pathsToTry;
 }
 
-// Not sure why we don't just return the full found path?
 export function getStrippedPath(tryPath: TryPath): string {
   return tryPath.type === "index"
     ? dirname(tryPath.path)
-    : tryPath.type === "file"
-    ? tryPath.path
-    : tryPath.type === "extension"
-    ? removeExtension(tryPath.path)
-    : tryPath.type === "package"
+    : tryPath.type === "file" ||
```

Comment:
This change affect other cases than when the file is coming from package.json fields (i.e the extensions case). You can see this by the fact the tests that are not releated to package.json needed changing. This change should probably be done too but in the interest of keeping the changes small I think we can wait with this change to later so this PR will only affect the case where the file is coming directly from package.json fields?

Reply:
It probably then makes sense that you directly change this stuff by yourself, this way you can implement each detail how you want it ;).

Reply:
My concern is that we may be breaking cases that we have not thought about. There are quite a few ppl using this package so the backlash may be quite large :-). So merging this PR will be easier the fewer cases we might be breaking. I'm thinking we can merge and publish this one change of how package.json main files are resolved and see if we get a lot of backlash for doing this. If it works out we can then look at making more changes, for example how extensions are resolved for regular files (not from package.json).

So if you could just revert the changes to `getStrippedPath()` and the related tests that would be great. I think your case will still be fixed by the remaining changes?

Reply:
I don't have more time to work on this ;). I am behind my own schedule since some time. So if you want more changes, go for it, everything is open source 😅.

> My concern is that we may be breaking cases that we have not thought about.
I respect this concern ^^! I have a long-term recommendation for you regarding this topic: too me it seems like this concern comes from the absence of integration tests, this makes it really hard for contributors to implement sensible changes!

Reply:
The concern actually comes from my very limited time to work on this :-) I don't have the time to handle any backlash so I want to minimize it as far as possible. Making a change is often quick and easy but maintaining the effects of the change can be really time consuming. Integration tests are great but having help maintaining the package is ever better. If you don't have the time to work on it I'll leave it open and see if someone else will pick it up.

@cspotcode Perhaps you want to continue the work on this?

Reply:
I think it's a chicken-and-egg problem.  Without proper documentation for how this library is supposed to behave, no one will want to help us maintain it.

For example, the `createMatchPath` docs don't explain how it resolves.  This leads me to assume they return absolute paths, but apparently that's not true because extensions get stripped off.  And now no one can remember why it does this.  If it returned absolute paths, my assumption would be correct, and no docs needed.  But the real behavior is strange, so it needs to be explained.

I think you and I will need to make the necessary behavior much clearer to both consumers and contributors.

Reply:
Yes, I agree it is not good that the rules for resolution is not clearer. In my view part of why the clarity is missing is becuase of the stripping of extensions and paths. A problem is that it already works for a lot of ppl so whatever we do could be breaking and we don't know since the rules are not explicitly documented. There are tests but this PR will break the tests. I think we should try do clear it up and release a major version so we can break things.

@cspotcode Do you think we close this PR then and continue work elsewhere?